docs/cli/function/layers.md

```diff
@@ -0,0 +1,219 @@
+
+You can either create a new function and add Lambda layers by running `amplify add function` or add layers to an existing function using `amplify update function`. Select `Lambda function` when prompted and you'll be presented the following question during the guided flow: 
+
+```console
+? Do you want to configure Lambda layers for this function? Yes
+? Provide existing layers or select layers in this project to access from this function (pick up to 5): (Press <space> to select, <a> to toggle all, <i> to invert selection)
+❯◯ Provide existing Lambda Layer ARNs
+ ◯ myamplifylayer1
+ ◯ myamplifylayer2
+```
+
+You can either add an existing layer in AWS by referencing its ARN or select a layer from your Amplify project that's listed below.
+
+When adding a layer from your Amplify project, you'll also be able to select the layer version. Amplify CLI automatically versions your Lambda layers by recognizing changes to the layer's folder contents. The largest layer version number represents the most recent changes. 
+```console
+? Select a version for myamplifylayer1:
+*> 8*
+  5
+  3
+  2
+  1
+```
+
+Given that layers can have overlapping contents, the order of the layer matters. You can adjust the layer's order if needed in the next step.
+
+```console
+? Modify the layer order:
+(Layers with conflicting files will overwrite contents of layers earlier in the list):
+- layer2
+- layer3
+- layer6
+- <ARN1>
+- <ARN2>
+```
+
+Now, you've successfully added a layer to your function.
+
+## Deploy Lambda layers & functions with Lambda layers
+
+Once you're ready with your changes in your layer and functions, you can deploy them by running `amplify push`. 
+
+If a Layer’s content has been updated and it has permissions associated, Amplify CLI will prompt you whether you want to carry the permissions forward to a newer version.
+
+```console
+Content changes in Lambda layer {LAYER_NAME} detected. What permissions do you want to grant to this new layer version?
+> The same permission as the latest layer version
+> Only accessible by the current account. You can always edit this later with: amplify update function
+```
+
+When a layer's content has been modified, Amplify CLI prompts you if the functions dependent on the layer should update to the latest layer version.
+
+```console
+You've updated Lambda layers: layer1, layer2...
+Do you want functions using the layers to upgrade to the latest version? (y/N)
+```
+
+## Update layer content
+
+Any file changes within a layer's folder are automatically tracked by Amplify CLI. If there are changes available, the Amplify CLI will create a new layer version with the changes and ask during `amplify push` if you'd like to update all dependent functions to use the latest version.
+
+You also have the ability to manually increment the layer version is added to a function by running `amplify update function` and selecting the `function` that you want to change the layer configuration for.
```

Comment:
This is not entirely true. A layer's version is only updated if runtimes change, or if content changes are detected in the layer folder